__tests__/ethereum-mock.js

```diff
@@ -11,7 +11,12 @@ export const _setupEthereumMock = ClientFunction((privateKey, chainId, rpcUrl) =
 });
 
 export const setupEthereumMock = async (privateKey, chainId) => {
-  await t.expect(ClientFunction(() => window.initialize)).ok({ timeout: FIXTURES_CONST.TIMEOUT });
+  await t
+    // Make sure to call the client function in order to assert against its return value:
+    // e.g "You passed a ClientFunction object to 't.expect()'.
+    // If you want to check the function's return value, use parentheses to call the function: fnName()."
+    .expect(ClientFunction(() => window.ethereum)())
```

Comment:
We want to check that `initialize` is available here, as `window.ethereum` has not necessarily been injected yet.

Reply:
`ClientFunction` return values are tricky.
Can't pass a reference to a window object to the caller, and cloning functionality inside the ClientFn are absent (Object.assign, JSON.stringify etc). 

I found a solution by passing a predicate to `waitForProperty`.